librepo/downloader.c

```diff
@@ -18,7 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-#define _XOPEN_SOURCE   500 // Because of fdopen() and ftruncate()
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
```

Comment:
Why did you drop `_XOPEN_SOURCE 500`?

Reply:
I need futimes() to change file timestamps and this requires _DEFAULT_SOURCE (for glibc > 2.19) or _BSD_SOURCE (for glibc <= 2.19, which is the case for RHEL7).
Having _XOPEN_SOURCE explicitly defined means, that _DEFAULT_SOURCE is not defined.

Reply:
How does this affect non-glibc environments?

Reply:
I'm not sure. But I can return _XOPEN_SOURCE   500 back

Reply:
It shouldn't hurt anything to put it back, and I'm wary of accidentally breaking non-glibc builds (there are a few out there, including Yocto...).